src/main/java/de/adesso/kicker/notification/NotificationService.java

```diff
@@ -1,56 +1,169 @@
 package de.adesso.kicker.notification;
 
-import java.util.ArrayList;
-import java.util.List;
-
+import de.adesso.kicker.notification.matchcreationrequest.MatchCreationRequestService;
+import de.adesso.kicker.notification.matchverificationrequest.MatchVerificationRequestService;
+import de.adesso.kicker.notification.teamjoinrequest.TeamJoinRequestService;
+import de.adesso.kicker.notification.tournamentjoinrequest.TournamentJoinRequestService;
+import de.adesso.kicker.user.User;
+import de.adesso.kicker.user.UserService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
-import de.adesso.kicker.user.User;
-import de.adesso.kicker.user.UserService;
+import java.util.ArrayList;
+import java.util.List;
 
 @Service
 public class NotificationService {
 
     private NotificationRepository notificationRepository;
+    private UserService userService;
+    private TeamJoinRequestService teamJoinRequestService;
+    private TournamentJoinRequestService tournamentJoinRequestService;
+    private MatchCreationRequestService matchCreationRequestService;
+    private MatchVerificationRequestService matchVerificationRequestService;
 
     @Autowired
-    public NotificationService(NotificationRepository notificationRepository) {
+    public NotificationService(NotificationRepository notificationRepository, UserService userService,
+            TeamJoinRequestService teamJoinRequestService, TournamentJoinRequestService tournamentJoinRequestService,
+            MatchCreationRequestService matchCreationRequestService,
+            MatchVerificationRequestService matchVerificationRequestService) {
+
         this.notificationRepository = notificationRepository;
+        this.userService = userService;
+        this.teamJoinRequestService = teamJoinRequestService;
+        this.tournamentJoinRequestService = tournamentJoinRequestService;
+        this.matchCreationRequestService = matchCreationRequestService;
+        this.matchCreationRequestService = matchCreationRequestService;
     }
 
-    // Gets a notification by id
     public Notification getNotificationById(long id) {
 
         return notificationRepository.findByNotificationId(id);
     }
 
-    // Get all notifications
     public List<Notification> getAllNotifications() {
 
         List<Notification> notifications = new ArrayList<>();
         notificationRepository.findAll().forEach(notifications::add);
         return notifications;
     }
 
-    // Get all notifications that were send by a specific user
     public List<Notification> getAllNotificationsBySender(User sender) {
```

Comment:
Is is necessary to have this overloaded?

Reply:
You meant the duplicate of the line:
"this.matchCreationRequestService = matchCreationRequestService;"
Fixed that.